Cython/Utility/Optimize.c

```diff
@@ -621,7 +625,11 @@ static double __Pyx__PyObject_AsDouble(PyObject* obj) {
         }
 #endif
         if (likely(float_value)) {
+#if CYTHON_ASSUME_SAFE_MACROS
             double value = PyFloat_AS_DOUBLE(float_value);
+#else
+            double value = PyFloat_AsDouble(float_value);
+#endif
```

Comment:
We do this all over, so let's add a `__Pyx_PyFloat_AS_DOUBLE()` macro for this. That also avoids the need for a second special case in the `__Pyx_PyObject_AsDouble()` macro (which isn't as efficient as it could be since it doesn't dispatch to `PyLong_AsDouble()` directly (or even indirectly)).

Also, we should decide whether the right guard is "Limited API" or "safe macros". I don't think there is anything unsafe about calling `PyFloat_AS_DOUBLE()` on a known Python `float` object, so I vote for "Limited API" as a special case.